app/src/main/java/com/nervousfish/nervousfish/data_objects/RSAKey.java

```diff
@@ -0,0 +1,62 @@
+package com.nervousfish.nervousfish.data_objects;
+
+/**
+ * RSA variant of {@link IKey}.
+ */
+public final class RSAKey implements IKey {
+
+    private final static String TYPE = "RSA";
```

Comment:
Yeah, I'm a bit in doubt as well. I thought as follows: imagine you read a key (so an IKey) from the database, then you want to know what kind of key it is. You'll probably use a few if statement or a switch statement to check if it's a simple or an RSA key. However, "if key == "simple"" is error-prone because a typing error results in an exception on runtime. If you use "if key == KEY_SIMPLE" you cannot make a typing mistake because Intellij will than yell that that variable doesn't exist